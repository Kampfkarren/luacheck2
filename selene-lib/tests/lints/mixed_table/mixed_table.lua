local bad = {
    "",
    a = b,
}

bad = {
    {},
    [a] = b,
}

bad = {
    a,
    [""] = b,
}

-- This is technically not a mixed table, but it's formatted like it harming readability
-- so it should still be linted
bad = {
    1,
    [2] = b,
}

bad = {
    [a] = b,
    [c] = d,
    "",
}

bad({
    a = b,
    "",
    c = d,
})

local good = {
    a = b,
    c = d,
}

good = {
    "",
    a,
}

good = {
    [1] = a,
    [3] = b,
}

good({
    a = b,
    c = d,
})
