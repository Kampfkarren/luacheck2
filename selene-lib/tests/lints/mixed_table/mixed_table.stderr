error[mixed_table]: mixed tables should be avoided, as they can cause confusing and hard to debug issues such as during iteration or encoding
  ┌─ mixed_table.lua:2:5
  │  
2 │ ╭     "",
3 │ │     a = b,
  │ ╰─────────^
  │  
  = help: change this table to either an array or dictionary

error[mixed_table]: mixed tables should be avoided, as they can cause confusing and hard to debug issues such as during iteration or encoding
  ┌─ mixed_table.lua:7:5
  │  
7 │ ╭     {},
8 │ │     [a] = b,
  │ ╰───────────^
  │  
  = help: change this table to either an array or dictionary

error[mixed_table]: mixed tables should be avoided, as they can cause confusing and hard to debug issues such as during iteration or encoding
   ┌─ mixed_table.lua:12:5
   │  
12 │ ╭     a,
13 │ │     [""] = b,
   │ ╰────────────^
   │  
   = help: change this table to either an array or dictionary

error[mixed_table]: mixed tables should be avoided, as they can cause confusing and hard to debug issues such as during iteration or encoding
   ┌─ mixed_table.lua:19:5
   │  
19 │ ╭     1,
20 │ │     [2] = b,
   │ ╰───────────^
   │  
   = help: change this table to either an array or dictionary

error[mixed_table]: mixed tables should be avoided, as they can cause confusing and hard to debug issues such as during iteration or encoding
   ┌─ mixed_table.lua:25:5
   │  
25 │ ╭     [c] = d,
26 │ │     "",
   │ ╰──────^
   │  
   = help: change this table to either an array or dictionary

error[mixed_table]: mixed tables should be avoided, as they can cause confusing and hard to debug issues such as during iteration or encoding
   ┌─ mixed_table.lua:30:5
   │  
30 │ ╭     a = b,
31 │ │     "",
   │ ╰──────^
   │  
   = help: change this table to either an array or dictionary

